src/plugins/discover/public/plugin.tsx

```diff
@@ -251,50 +247,8 @@ export class DiscoverPlugin
       ),
     });
 
-    const {
-      appMounted,
-      appUnMounted,
-      stop: stopUrlTracker,
-      setActiveUrl: setTrackedUrl,
-      restorePreviousUrl,
-    } = createKbnUrlTracker({
-      // we pass getter here instead of plain `history`,
-      // so history is lazily created (when app is mounted)
-      // this prevents redundant `#` when not in discover app
-      getHistory: getScopedHistory,
-      baseUrl,
-      defaultSubUrl: '#/',
-      storageKey: `lastUrl:${core.http.basePath.get()}:discover`,
-      navLinkUpdater$: this.appStateUpdater,
-      toastNotifications: core.notifications.toasts,
-      stateParams: [
-        {
-          kbnUrlKey: '_g',
-          stateUpdate$: plugins.data.query.state$.pipe(
-            filter(
-              ({ changes }) => !!(changes.globalFilters || changes.time || changes.refreshInterval)
-            ),
-            map(({ state }) => ({
-              ...state,
-              filters: state.filters?.filter(esFilters.isFilterPinned),
-            }))
-          ),
-        },
-      ],
-      onBeforeNavLinkSaved: (newNavLink: string) => {
-        // Do not save SEARCH_SESSION_ID into nav link, because of possible edge cases
-        // that could lead to session restoration failure.
-        // see: https://github.com/elastic/kibana/issues/87149
-        if (newNavLink.includes(SEARCH_SESSION_ID_QUERY_PARAM)) {
-          newNavLink = replaceUrlHashQuery(newNavLink, (query) => {
-            delete query[SEARCH_SESSION_ID_QUERY_PARAM];
-            return query;
-          });
-        }
-
-        return newNavLink;
-      },
-    });
+    const { setTrackedUrl, restorePreviousUrl, stopUrlTracker, appMounted, appUnMounted } =
```

Comment:
Why did I refactor this function to a separate file. So I had the issue that the the Pageload size increased in my intial commit, so I tried to refactor the part of` plugin.tsx` using `esFilters.isFilterPinned` to it's own file importing it async on application mount. However this caused troubles with kbn url tracking, so I decided a different size. However, due to the size of `plugin.tsx` I think it's good to extract this function to clean up code a bit.